lib/aws/rails/sqs_active_job/configuration.rb

```diff
@@ -25,6 +25,7 @@ class Configuration
         DEFAULTS = {
           max_messages: 10,
           shutdown_timeout: 15,
+          retry_standard_errors: true,
```

Comment:
Do we really need to preserve backwards compatibility for this? It seems safe to not rely on the fact that something failed and retried implicitly.

Reply:
I added retry_standard_errors with a default of true because right now, users are getting a built in retry of everything and they could be relying on that to retry expected, transient errors.  If they set it to false (or if we default it to false), any of those errors that were previously retried will now fail immediately on the first error.

I've also added a check on `execution_exceptions` and we will NOT retry if ActiveJob has already caught/handled exceptions (note, this is true both for `retry_on` and `discard_on`).  I think this is a reasonable compromise that address the original issue while not removing existing retry behavior that users may be relying on.